### PR TITLE
Python: Add support for falcon web API framework.

### PR DESCRIPTION
**Pull request by markshannon**

### CODE REVIEW

File: python/ql/src/semmle/python/security/strings/External.qll [hunk 1]
```diff
@@ -96,3 +96,23 @@ private predicate json_load(ControlFlowNode fromnode, CallNode tonode) {
     )
 }
 
+/** A kind of "taint", representing am open file-like object from an external source. */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/security/strings/External.qll]: Typo: `am -> an`

File: python/ql/src/semmle/python/security/strings/External.qll [hunk 2]
```diff
@@ -96,3 +96,23 @@ private predicate json_load(ControlFlowNode fromnode, CallNode tonode) {
...
+class ExternalFileObject extends TaintKind {
+
+    ExternalFileObject() {
+        this = "file[" + any(ExternalStringKind key) + "]"
+    }
+
+
+    /** Gets the taint kind for item in this sequence */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/security/strings/External.qll]: This does not seem to match what the predicate does.

File: python/ql/src/semmle/python/web/falcon/Request.qll [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+/** https://falcon.readthedocs.io/en/stable/api/request_and_response.html */
+class FalconRequest extends TaintKind {
+
+    FalconRequest() {
+        this = "falcon.request"
+    }
+
+    override TaintKind getTaintOfAttribute(string name) {
+        // name = "env" and result instanceof WsgiEnvironment
```
[the remark this answers is not in the document]
Comment (markshannon) [reply python/ql/src/semmle/python/web/falcon/Request.qll]: Yes, I've added a `WsgiEnvironment` taint kind.

File: python/ql/src/semmle/python/security/strings/External.qll (hunk 1 above)
Comment (markshannon) [reply python/ql/src/semmle/python/security/strings/External.qll]: done

File: python/ql/src/semmle/python/security/strings/External.qll (hunk 2 above)
Comment (markshannon) [reply python/ql/src/semmle/python/security/strings/External.qll]: fixed

**Pull request merged**